src/App.tsx

```diff
@@ -15,47 +16,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABETICALLY = 'alphabetically',
+  SORT_BY_LENGTH = 'sortByLength',
+}
+
+function sortedGoodByType(goods: string[],
+  sortType: string,
+  isReversed: boolean): string[] {
+  const copyGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABETICALLY:
+      copyGoods.sort((good1, good2) => (
+        good1.localeCompare(good2)
+      ));
+      break;
+
+    case SortType.SORT_BY_LENGTH:
+      copyGoods.sort((good1, good2) => (
+        good1.length - good2.length
+      ));
+      break;
+
+    default:
+      break;
```

Comment:
Do not leave the `default` block empty, you can add there, for example, an error about the wrong type of sorting